src/back-end/lib/resources/feedback.ts

```diff
@@ -0,0 +1,49 @@
+import { getString } from 'back-end/../shared/lib';
+import { AvailableModels, Session, SupportedRequestBodies } from 'back-end/lib/app/types';
+import * as crud from 'back-end/lib/crud';
+import { basicResponse, JsonResponseBody, makeJsonResponseBody, Response } from 'back-end/lib/server';
+import { CreateRequestBody, CreateValidationErrors } from 'shared/lib/resources/feedback';
+
+type CreateResponseBody = JsonResponseBody<null | CreateValidationErrors>;
+
+type RequiredModels = 'Feedback' | 'User';
+
+export type Resource = crud.Resource<SupportedRequestBodies, JsonResponseBody, AvailableModels, RequiredModels, CreateRequestBody, null, Session>;
+
+export const resource: Resource = {
+
+  routeNamespace: 'feedback',
+
+  create(Models) {
+    const FeedbackModel = Models.Feedback;
+    return {
+      async transformRequest(request) {
+        return {
+          rating: getString(request.body.value, 'rating'),
+          text: getString(request.body.value, 'text')
+        }
+      },
+      async respond(request): Promise<Response<CreateResponseBody, Session>> {
+        const respond = (code: number, body: null | CreateValidationErrors) => basicResponse(code, request.session, makeJsonResponseBody(body));
+
+        // Create feedback object
+        const feedback = new FeedbackModel({
+          createdAt: new Date(),
+          text: request.body.text,
+          rating: request.body.rating
+        });
+
+        // If we have an authenticated user, store the type with the feedback
+        if (request.session && request.session.user) {
+          feedback.userType = request.session.user.type;
+        }
+
+        await feedback.save();
+
+        return respond(200, null);
```

Comment:
Program staff should be notified of the feedback via email. Have a look at the `notifications.ts` module.

Reply:
Had a chat with Liz & co. Notification should be emailed to `Procurement.Concierge@gov.bc.ca` only, not all program staff.

Reply:
Also, we should follow the convention of other CRUD resources, and respond with the created resource encoded as JSON. In this case, you would need to export a `PublicFeedback` interface from the shared feedback resource module, and create a serialization function in the Feedback schema, and use the type in this module and the front-end's `http/api.ts` module.